src/fides/api/api/v1/endpoints/privacy_preference_endpoints.py

```diff
@@ -684,6 +700,32 @@ def validate_served_record(
             )
 
 
+def update_request_with_decoded_tc_string_fields(
+    request_body: PrivacyPreferencesRequest, db: Session
+) -> PrivacyPreferencesRequest:
+    """Update the request body with the decoded values of the TC string if applicable"""
+    if request_body.tc_string:
+        tcf_contents: TCFExperienceContents = get_tcf_contents(db)
```

Comment:
Comparing the TCF experience to the TC string is the only definitive way to know for which preferences we should save an explicit opt out, as the string only encodes opt-in.  This is not great for performance, but we have plans to cache this experience I believe. In the meantime, I only load this if a TC string is supplied in the request.